components package added

asanthamax opened the pull request:
refactor the package name to org.wso2.carbon.identity.user.store.mongodb and added with pom.xml file inside components folder

asanthamax committed:
src package and pom file added

asanthamax committed:
modified the package structure

cdwijayarathna commented:
You need to add wso2 license headers to the files

asanthamax commented:
@cdwijayarathna thanks for the comment can you explain me  where can i find wso2 license header to add?

omindu commented:
Check files in https://github.com/wso2-extensions/identity-inbound-auth-oauth

asanthamax committed:
license header added

asanthamax committed:
license header added

asanthamax committed:
license header added

asanthamax committed:
license header added

asanthamax committed:
license header added

asanthamax committed:
license header added

asanthamax committed:
license header added

asanthamax committed:
license header added

asanthamax committed:
license header added

asanthamax committed:
license header added

asanthamax committed:
license header added

asanthamax commented:
@omindu thanks for the link i have updated the files with license headers

cdwijayarathna commented:
The year in the license need to be 2016,

asanthamax committed:
license headers modified change year to 2016

asanthamax commented:
@cdwijayarathna i have changed the year to 2016 in license header

asanthamax committed:
features and parent pom.xml added

thariyarox commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,899 @@
+   // private List<DBObject> queryList = null;
+   // private List<DBObject> projectionList = null;
+    private BulkWriteOperation bulkWrite = null;
+	private  boolean multipleLookUp = false;
```
formatting issue

thariyarox commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,899 @@
+    //private static boolean dependencyTrue = false;
+    private String distinctKey = "";
+    private boolean isCaseSensitive = true;
+	private Map<String,Object> mapMatchCaseInSensitive = null;
```
formatting issue

thariyarox commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,899 @@
+			mapMatch = new HashMap<String, Object>();
+		}
+        this.multiMapLookup = new ArrayList<Map<String, Object>>();
+        this.multiMapUnwind = new ArrayList<Map<String, Object>>();
```
fromatting issue

thariyarox commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,899 @@
+		this.projection = null;
+		this.parameterCount = 0;
+        this.distinctKey = "";
+        this.isCaseSensitive = true;
```
formatting issue

thariyarox commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,899 @@
+		this.mapProject =null;
+		this.mapSort = null;
+		this.mapGroup = null;
+		this.mapUnwind = null;
```
formatting issue

thariyarox commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/userstoremanager/MongoDBRealmConstants.java:
```diff
@@ -0,0 +1,153 @@
+	    public static final String USER_NAME_UNIQUE_MONGO_QUERY = "{'collection' : 'UM_USER','UM_USER_NAME' : '?','projection' : {'UM_ID' : '1','_id' : '0'}}";
+	    public static final String IS_DOMAIN_EXISTS_MONGO_QUERY = "{'collection' : 'UM_DOMAIN','UM_DOMAIN_NAME' : '?','UM_TENANT_ID' : '?','projection' : {'UM_DOMAIN_ID' : '1'}}";
+        public static final String ADD_DOMAIN_MONGO_QUERY = "{'collection' : 'UM_DOMAIN','UM_DOMAIN_NAME' : '?','UM_TENANT_ID' : '?','UM_DOMAIN_ID' : '?'}";
+		public static final String GET_SHARED_ROLES_FOR_USER_MONGO_QUERY = "{'collection' : 'UM_SHARED_USER_ROLE','user.UM_USER_NAME' : '?','UM_USER_TENANT_ID' : '?','UM_USER_TENANT_ID' : 'user.UM_TENANT_ID','UM_ROLE_TENANT_ID' : 'role.UM_TENANT_ID','$lookup' : [{'from' : 'UM_USER','localField' : 'UM_USER_ID','foreignField' : 'UM_ID','as' : 'user'},{'from' : 'UM_ROLE','localField' : 'UM_ROLE_ID','forei ...
```
formatting issue

thariyarox commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/userstoremanager/MongoDBRealmConstants.java:
```diff
@@ -0,0 +1,153 @@
+	    public static final String URL = "url";
+	    public static final String USER_NAME = "userName";
+	    public static final String PASSWORD = "password";
+		public static final String PORT = "port";
```
formatting issue

asanthamax committed:
features added and code reformatted

asanthamax committed:
pom files changed

asanthamax committed:
export package corrected

asanthamax committed:
json lib version changed to 20140107

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/pom.xml:
```diff
@@ -0,0 +1,99 @@
+                            org.wso2.carbon.user.api; version="${carbon.user.api.imp.pkg.version.range}",
+                            *;resolution:=optional
+                        </Import-Package>
+                        <DynamicImport-Package>*</DynamicImport-Package>
```
Remove dynamic imports if you are not doing any dynamic class loading in the component.

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/pom.xml:
```diff
@@ -0,0 +1,99 @@
+                            org.osgi.service.component; version="${osgi.service.component.imp.pkg.version.range}",
+                            org.wso2.carbon.user.core.*;version="${carbon.kernel.package.import.version.range}",
+                            org.wso2.carbon.user.api; version="${carbon.user.api.imp.pkg.version.range}",
+                            *;resolution:=optional
```
Better to remove the "*;resolution:=optional" and include each imported package separately

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,874 @@
+    private DBObject query = null;
+    private DBObject projection;
+    private String defaultQuery;
+    private HashMap<String, Object> parameterValue;
```
Use Map instead of HashMap

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,874 @@
+    }
+
+    public WriteResult insert() throws MongoQueryException {
+        // TODO Auto-generated method stub
```
Remove the TODO's and auto generated lines

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,874 @@
+
+            return this.collection.aggregate(pipeline);
+        } catch (MongoException e) {
+
```
remove these extra lines

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,874 @@
+
+
+    public WriteResult update(boolean upsert, boolean multi) throws MongoQueryException {
+        // TODO Auto-generated method stub
```
Remove the TODO's

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,874 @@
+
+    public WriteResult update(boolean upsert, boolean multi, WriteConcern aWriteConcern, DBEncoder encoder)
+            throws MongoQueryException {
+        // TODO Auto-generated method stub
```
Remove the TODO's

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,874 @@
+
+    public WriteResult update(boolean upsert, boolean multi, WriteConcern aWriteConcern,
+                              boolean byPassDocumentValidation, DBEncoder encoder) throws MongoQueryException {
+        // TODO Auto-generated method stub
```
Remove the TODO's

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/userstoremanager/internal/MongoDBUserStoreDSComponent.java:
```diff
@@ -0,0 +1,65 @@
+        cc.getBundleContext().registerService(UserStoreManager.class.getName(), userStoreManager, null);
+        log.info("MongoDB User Store bundle activated successfully..");
+        UserStoreManagerRegistry.init(cc.getBundleContext());
+        System.out.println("Mongo Started");
```
Use log.info instead

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/util/MongoDatabaseUtil.java:
```diff
@@ -0,0 +1,993 @@
+        } catch (MongoQueryException ex) {
+
+            log.error(ex.getMessage(), ex);
+            log.error("Using json : " + stmt);
```
Do not log same error in 2 statements. May get mixed with other logs.

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/util/MongoDatabaseUtil.java:
```diff
@@ -0,0 +1,993 @@
+        } catch (MongoQueryException ex) {
+
+            log.error(ex.getMessage(), ex);
+            log.error("Using json : " + stmt);
```
Do not log same error in 2 statements. May get mixed with other logs.

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/util/MongoDatabaseUtil.java:
```diff
@@ -0,0 +1,993 @@
+            }
+        } catch (MongoQueryException ex) {
+            log.error("Error! " + ex.getMessage(), ex);
+            log.error("Using json " + stmt);
```
Do not log same error in 2 statements. May get mixed with other logs.

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/util/MongoDatabaseUtil.java:
```diff
@@ -0,0 +1,993 @@
+            }
+        } catch (MongoQueryException ex) {
+            log.error("Error! " + ex.getMessage(), ex);
+            log.error("Using json " + stmt);
```
Do not log same error in 2 statements. May get mixed with other logs.

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/util/MongoDatabaseUtil.java:
```diff
@@ -0,0 +1,993 @@
+                }
+            }
+            return values;
+        } catch (NullPointerException ex) {
```
Any reason for catching NPEs? Better handle them at code

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/util/MongoDatabaseUtil.java:
```diff
@@ -0,0 +1,993 @@
+                }
+            }
+            return values;
+        } catch (NullPointerException ex) {
```
Any reason for catching NPEs? Better handle them at code

pulasthi7 commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/userstoremanager/internal/MongoDBUserStoreDSComponent.java:
```diff
@@ -0,0 +1,65 @@
+
+    @SuppressWarnings({"RedundantThrows", "UnusedParameters"})
+    protected void deactivate(ComponentContext cc) throws Exception {
+        System.out.println("MongoDB Bundle Shutting down");
```
Use log.info instead

asanthamax committed:
issues fixed

asanthamax commented:
@pulasthi7  thanks for the feedback i have corrected the above issues and i have added new orbit to add  org.json new version cuz older version in IS causing some errors to user activities after adding version 20160212 as orbit issue fixed

asanthamax committed:
Delete MANIFEST.MF

asanthamax committed:
org.json added to bundle definiton

omindu commented on components/org.wso2.carbon.identity.user.store.mongodb/pom.xml:
```diff
@@ -0,0 +1,95 @@
+                            org.wso2.carbon.user.core.*;version="${carbon.kernel.package.import.version.range}",
+                            org.wso2.carbon.user.api; version="${carbon.user.api.imp.pkg.version.range}",
+                        </Import-Package>
+                        <DynamicImport-Package>*</DynamicImport-Package>
```
@thariyarox Is this needed ?

omindu commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatement.java:
```diff
@@ -0,0 +1,310 @@
+import org.wso2.carbon.user.api.UserStoreException;
+
+/**
+ * MongoDB Prepared Statement Interface
```
End class level and method level comments with a period.

omindu commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatement.java:
```diff
@@ -0,0 +1,310 @@
+public interface MongoPreparedStatement {
+
+    /**
+     * set int parameter value to respective query parameter
```
Capitalize first letter class level and method level comments and end with a period. Check and fix in other places as well.

omindu commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,768 @@
+                        for (Map<String, Object> map : multiMapLookup) {
+                            if (map.containsKey("dependency")) {
+                                map.remove("dependency");
+                                DBObject lookup = new BasicDBObject("$lookup", new BasicDBObject(map));
```
You can use String constants if the Strings are getting duplicated in multiple places.

omindu commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,768 @@
+                    match = new BasicDBObject("$match", new BasicDBObject(mapMatch));
+                } else {
+
+                    match = new BasicDBObject("$match", new BasicDBObject(mapMatch).append("UM_USER_NAME", new BasicDBObject(mapMatchCaseInSensitive)));
```
Check line width.

omindu commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,768 @@
+            throw new MongoQueryException("Parameter count not matched with query parameters");
+        } else {
+            if (convertToDBObject(defaultQuery)) {
+                return this.collection.update(this.query, new BasicDBObject("$set", this.projection), upsert, multi, aWriteConcern);
```
Check line width.

omindu commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,768 @@
+            throw new MongoQueryException("Parameter count not matched with query parameters");
+        } else {
+            if (convertToDBObject(defaultQuery)) {
+                return this.collection.update(this.query, new BasicDBObject("$set", this.projection), upsert, multi, aWriteConcern, encoder);
```
Check line width. Fix all the places where linewidth exceeds 120 characters.

omindu commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,768 @@
+        try {
+            return this.bulkWrite.execute();
+        } catch (Exception e) {
+            throw new MongoQueryException("Query Exception:" + e.getLocalizedMessage());
```
Is there a reason why `e` is not wrapped in `MongoQueryException` ?

omindu commented on components/org.wso2.carbon.identity.user.store.mongodb/src/main/java/org/wso2/carbon/identity/user/store/mongodb/query/MongoPreparedStatementImpl.java:
```diff
@@ -0,0 +1,768 @@
+     */
+    private boolean convertToDBObject(String query) {
+        JSONObject queryObject = new JSONObject(query);
+        if (queryObject.has("collection")) {
```
Can use a String constant for "collection".

omindu commented:
Exception handling should be improved. Please have a chat with Tharindu and update the PR.

asanthamax committed:
Delete pom.xml

cdwijayarathna commented:
It would be better if you can squash all commits into 1 commit with a meaningful name which describe what you have done. Is there a public jira for this feature? If so please use it in the PR name and commit message so it will be easy to refer. If there is no public jira, please create a new one (I'm not sure if this should go under Identity server project or extensions project).

asanthamax committed:
budel definiton modified and code format issues fixed

asanthamax committed:
components pom.xml updated

asanthamax committed:
update comments start letter to capital

asanthamax committed:
fixed code format issues and exception handling improved

mefarazath commented:
@thariyarox  Can we merge this now?

JKAUSHALYA commented:
Closing since not approved.
